rasterio/_copy.pyx

```diff
@@ -1,48 +1,76 @@
 """Raster copying."""
 
+
 include "gdal.pxi"
 
 import logging
-import os
-import os.path
 
-from rasterio._err cimport exc_wrap_int, exc_wrap_pointer
+from rasterio._io cimport DatasetReaderBase
+from rasterio._err cimport exc_wrap_pointer
+from rasterio.env import ensure_env
 
 
 log = logging.getLogger(__name__)
 
 
-cdef class RasterCopier:
+@ensure_env
+def copy(src, dst, driver='GTiff', strict=False, **creation_options):
```

Comment:
@sgillies Is there any reason not to do this?  Significant slowdown on `import rasterio`?  A performance hit in a function that mixes Python and Cython code that is greater than just copying data between Python and Cython?  Previously we had a `rasterio.__init__.copy()` function that called `rasterio._copy.RasterCopier()`, but it seems like we don't really need the Python layer and can just expose the Python function directly.